smart_home/include/devices/device.h

```diff
@@ -0,0 +1,24 @@
+#ifndef SMART_HOME_DEVICES_DEVICE_H_
+#define SMART_HOME_DEVICES_DEVICE_H_
+
+#include "../rooms/room.h"
+
+namespace devices {
+
+// Rule of Five
+
+class Device {
+  public:
+    virtual rooms::Room* room() = 0;
+
+  protected:
+    Device(rooms::Room* room): room_(room) {}
+
+    virtual ~Device();
+
+    rooms::Room* room_;
```

Comment:
as a best practice and security mitigation, it's best to set all pointers to nullptr by default if you cannot assign a value right away.